src/transformState.js

```diff
@@ -5,7 +5,39 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    reducer(action);
+  }
+
+  function reducer({ type, extraData, keysToRemove }) {
+    switch (type) {
+      case 'addProperties':
+        Object.assign(state, extraData);
+
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          if (state.hasOwnProperty(key)) {
+            delete state[key];
+          }
```

Comment:
No need for this check here, `delete` works fine for non existent properties.
```suggestion
          delete state[key];
```

Reply:
E.g. no errors here:
```js
const a = {};
delete a["b"];
```